### ISSUE TITLE
Switch Default Backend to JAX.

### ISSUE THREAD

**Issue by Thenerdstation** [google/TensorNetwork#183]
Let's face it, TensorFlow is too difficult to use for new users and people outside of the machine learning community (and even people within the ML community!). It's preventing people from using our library simply because trying to do stuff with TF is way too hard.

We should switch the JAX as the default. It's cleaner, easier to use, and closer to the existing tooling that scientists already use (i.e, numpy). This will be a breaking change, but we did warn that we would do that.

**Comment by amilsted** [comment]
Eeep. Isn't JAX still a little rough around the edges for this? If we switch away from TF as default, I would vote for numpy, which a large portion of the scientific community already use.

**Comment by Thenerdstation** [comment]
It's actually really good at this point. A lot of Google researchers have been using it, and bugs get fixed pretty quickly.

You should try it! I'd recommend trying to port one of your TF models.

**Comment by Thenerdstation** [comment]
As per the numpy backend, default settings set the expectation for a library, so if we default to numpy then this will effectively be a "numpy first" library and a lot of users will miss out on the awesome advantages of JAX like autodiff and GPU/TPU support.

I also think we should start getting scientists to use JAX just in general. If it's the default, then researchers will have a stronger incentive to use it.

**Comment by Thenerdstation** [comment]
Plus, JAX is definitely more stable than TF 2.0, and Pytorch doesn't support complex numbers.

**Comment by amilsted** [comment]
Well, I just upgraded my jax and I'm glad to see the edges have gotten rather smoother: There are no longer scary warnings in the readme about using the PIP wheels, or runtime warnings about using linalg methods! This makes me happier about it becoming the default.

**Comment by amilsted** [comment]
Yeah, I think you have my vote now. JAX seems to be improving and maturing very quickly and it is a much lighter dependency than TF. Also, it is much easier to transition from numpy to JAX than to TF!

**Comment by Thenerdstation** [comment]
We have gotten the OK from @sleichen to do the switch. We will officially push the switch in the first week of September as version `0.1.0`

**Comment by amilsted** [comment]
Lack of windows support in JAX might be a barrier to a number of physicists. Let's make sure it's easy to use TensorNetwork with numpy or TF even if JAX can't be installed.

**Comment by Thenerdstation** [comment]
Yeah, @sleichen and I talked about it yesterday. We've decided to make the default numpy, as that is what people generally expect when they first try the library. And in the tutorials we will push people to move to start using JAX.
